### PR TITLE
[stable27] The almighty text processing PR

### PR DESCRIPTION
**Pull request by marcelklehr**
Contains backports of 

 - [x] https://github.com/nextcloud/server/pull/38854
 - [x] https://github.com/nextcloud/server/pull/39567
 - [x] https://github.com/nextcloud/server/pull/39684
 - [x] tests fix from https://github.com/nextcloud/server/pull/39660
 - [x] https://github.com/nextcloud/server/pull/39680
- bumps version to 27.1.0.0

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Review by julien-nc — APPROVED** [review]
:rocket:

**Comment by julien-nc** [comment]
Psalm issue because missing `core/ResponseDefinitions.php`. Even if we put add, the `OCA\Core` namespace is not valid in stable27.
@provokateurin Any clue?

**Comment by provokateurin** [comment]
The annotations are not available in stable27. The are only on master

**Comment by marcelklehr** [comment]
> The annotations are not available in stable27.

Ah, right. then we need to remove them

**Comment by marcelklehr** [comment]
:tada:

**Review by juliusknorr — APPROVED** [review]
Looks good from my side, just left a comment on the version bump to clarify before merge

**Comment by julien-nc** [comment]
Removed the 2 commits changing version.php, added one bumping the last version digit. Put the "compile assets" commit on top.

File: version.php [hunk 1]
```diff
@@ -30,10 +30,10 @@
 // between betas, final and RCs. This is _not_ the public version number. Reset minor/patch level
 // when updating major/minor version number.
 
-$OC_Version = [27, 0, 2, 1];
+$OC_Version = [27, 1, 0, 0];
 
 // The human-readable string
-$OC_VersionString = '27.0.2';
+$OC_VersionString = '27.1.0 dev';
```
[the remark this answers is not in the document]
Comment (skjnldsv) [reply version.php]: Nope it should not! Pr is in #39757

**Pull request merged**